rotkehlchen/tests/api/test_reports.py

```diff
@@ -0,0 +1,196 @@
+            notes='Send 0.5 ETH to 0xABC',
+            location=Location.ETHEREUM,
+            timestamp=timestamp_2_secs,
+            asset=A_ETH,
+            free_amount=ZERO,
+            taxable_amount=half_amount,
+            price=Price(eth_price_ts_2),
+            pnl=PNL(taxable=half_amount, free=ONE),
+            cost_basis=None,
+            index=1,
+            extra_data={},
+        ), ProcessedAccountingEvent(
+            event_type=AccountingEventType.TRANSACTION_EVENT,
+            notes='Received 100 DAI',
+            location=Location.ETHEREUM,
+            timestamp=timestamp_2_secs,
+            asset=A_DAI,
+            free_amount=hundred,
+            taxable_amount=ZERO,
+            price=Price(ONE),
+            pnl=PNL(taxable=ZERO, free=hundred),
+            cost_basis=None,
+            index=0,
+            extra_data={},
+        ),
+    ]
+
+    for event in events:
+        dbreport.add_report_data(
+            report_id=report_id,
+            time=event.timestamp,
+            ts_converter=timestamp_to_date,
+            event=event,
+        )
+
+    return report_id, events
+
+
+@pytest.mark.parametrize('have_decoders', [True])
+@pytest.mark.parametrize('mocked_price_queries', [prices])
+@pytest.mark.parametrize('initialize_accounting_rules', [True])
+def test_get_report_data(
+        rotkehlchen_api_server: APIServer,
+) -> None:
+    """Test that getting report data works correctly"""
+    # First create a report and added events
+    report_id, events = setup_report_events(rotkehlchen_api_server.rest_api.rotkehlchen.data.db)
+
+    # Query report data with no filters
+    response = requests.post(
+        api_url_for(rotkehlchen_api_server, 'per_report_data_resource', report_id=report_id),
+        json={},
+    )
+    assert_proper_response(response)
+    data = response.json()
+    assert data['message'] == ''
+    assert 'entries' in data['result']
+    assert 'entries_found' in data['result']
+    assert 'entries_total' in data['result']
+    assert 'entries_limit' in data['result']
+    assert len(data['result']['entries']) == len(events)
+    assert data['result']['entries_found'] == len(events)
+    assert data['result']['entries_total'] == len(events)
+    assert data['result']['entries_limit'] <= FREE_PNL_EVENTS_LIMIT
+
+    # Test pagination limits for free users
+    response = requests.post(
+        api_url_for(rotkehlchen_api_server, 'per_report_data_resource', report_id=report_id),
+        json={'offset': 0, 'limit': 1},
+    )
+    assert_proper_response(response)
+    data = response.json()
+    assert len(data['result']['entries']) == 1
+    assert data['result']['entries_found'] == 1
+    assert data['result']['entries_total'] == len(events)
+    assert data['result']['entries_limit'] == 1
+
+
+@pytest.mark.parametrize('start_with_valid_premium', [True])
+def test_get_report_data_with_premium(
+        rotkehlchen_api_server: APIServer,
+) -> None:
```

Comment:
unless I'm missing something you can use

```suggestion
@pytest.mark.parametrize('start_with_valid_premium', [True, False])
def test_get_report_data_with_premium(
        rotkehlchen_api_server: APIServer,
        start_with_valid_premium: bool,
) -> None:
```

and avoid repeating the test

Reply:
Ah ok thanks